### ISSUE TITLE
“Waiting on upstream data to be up to date” should let you see lineage to what you’re waiting on

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#18397]
Scenario: your asset isn't being auto-materialized because one of its parents is out of date with respect to one of its ancestors. In the asset-level auto-materialize history UI, you can get a link to that ancestor.

![image](https://github.com/dagster-io/dagster/assets/654855/25300e4e-805c-4741-9584-f48029671c1c)

But in many cases, you won't even be aware of the existence of that ancestor or how it relates to the asset you expect to be auto-materialized.

It would be helpful if we could link you to a filtered view of the asset graph that shows lineage between the node being considered for auto-materialization and the updated ancestor. This will allow the user to see the chain of nodes that have the yellow "Upstream data" tag and make it clear what needs remedying.

**Comment by mjkanji** [comment]
On a related note, it would be great if the query filter language could be expanded to would allow you to achieve the same in the global asset lineage. For example, `A => B` should show the path from A to B. 

While this is somewhat achievable if you show different levels of ancestors/children from a reference asset, it has the following problems: 

1. You have to know the degrees of separation between the two
2. It shows all the siblings of all the assets in the path, resulting in a much wider/more cluttered graph than just the direct path between A and B.

**Comment by sryza** [comment]
@mjkanji agreed that this would be useful. I think it's worth filing a separate issue for that, because might get lost in the comment for this issue.

**Comment by bengotow** [comment]
This came up again today in relation to #20251 - moving to our Linear as well https://linear.app/dagster-labs/issue/FE-317/waiting-on-upstream-data-to-be-up-to-date-should-let-you-see-lineage
